core/include/cubos/core/ecs/system.hpp

```diff
@@ -70,6 +70,162 @@ namespace cubos::core::ecs
         SystemInfo mInfo; ///< Information about the wrapped system.
     };
 
+    /// System argument which provides read access to the given resource or component.
+    /// Can be used as a pointer with both the `->` and `.` operators.
+    template <typename T>
+    class Read
+    {
+    public:
+        /// Creates a new read argument.
+        /// @param ref Reference to the resource or component.
+        inline Read(const T& ref)
+            : ref(ref)
+        {
+        }
+
+        /// @returns A pointer to the resource or component.
+        inline const T* operator->() const
+        {
+            return &this->ref;
+        }
+
+        /// @returns A reference to the resource or component.
+        inline const T& operator*() const
+        {
+            return this->ref;
+        }
+
+        /// @returns A reference to the resource or component.
+        inline const T& get() const
+        {
+            return this->ref;
+        }
```

Comment:
Is it ok to have these two methods of getting the reference? It might lead to inconsistency and make it harder to read the codebase.

Reply:
True, lets keep only the dereference then.